packages/datadog-plugin-google-cloud-pubsub/src/consumer.js

```diff
@@ -30,6 +30,8 @@ class GoogleCloudPubsubConsumerPlugin extends ConsumerPlugin {
   finish (message) {
     const span = this.activeSpan
 
+    if (!span) return
```

Comment:
This would result in the span and thus the trace never finishing. Do you know why the span is missing in the first place? It should always be active where the subscription is closed.

Reply:
I understand that this means this particular span will not be recorded, but I'm not sure how this can be avoided. Is there a way to access all inflight messages from this subscription? Is it worth the trouble?

The missing span kind of makes sense to me - the subscription is closed in a different context from the message consumption. There might be several inflight messages being processed at this point. You can see an example in the test case I added.

If this doesn't makes sense to you - could you direct me to the correct place in the code to investigate this?

Reply:
The problem with never finishing the span is that this completely hides that this happened in the first place. In general we try to never leave a span unfinished for that reason.

Is there an event emitter when this happens? If yes then it could be captured on the instrumentation side. Otherwise the `close` method of the subscription could potentially be patched.

Reply:
@rochdev is this going to be addressed as part of a different PR by DD team? We are getting the same issue. And not having a clean exit for us is a much bigger problem than missing a span which is happened during the shutdown sequence

Reply:
Do you have a proper reproduction case? The test in this PR seems to pass _without_ the change so we would need some case where the fix can actually be _proven_. 🤔

Reply:
Looks like the original issue was fixed in #3617 by checking for the span directly in `finish`. As mentioned above, this does mean that a trace could be prevented from finishing, but I tend to agree that it's more important not to crash, and we could track unfinished spans through telemetry instead.

There can be an additional issue when the span is missing and the message has been handled which is not covered by the tests of this PR, but the PR itself does fix that other issue and it's unclear how it could be tested in the first place. So I will go ahead and merge this.